cpp/oneapi/dal/backend/primitives/optimizers/common_dpc.cpp

```diff
@@ -90,6 +90,33 @@ sycl::event l1_norm(sycl::queue& queue,
     });
 }
 
+template <typename Float>
+sycl::event max_abs(sycl::queue& queue,
```

Comment:
There are some existing primitive functions that could possibly be equivalent to this. Not sure if you've checked those out. Source: https://github.com/oneapi-src/oneDAL/blob/main/cpp/oneapi/dal/backend/primitives/reduction/reduction_dpc.cpp, example call: https://github.com/oneapi-src/oneDAL/blob/main/cpp/oneapi/dal/algo/svm/backend/gpu/train_results.hpp#L114-L119

Reply:
These primitives work take 2d-data as an input while we need to reduce a 1d-vector. I think it is unreasonable to cast data to 2d ndview in order to use these functions.